code/weapon/beam.cpp

```diff
@@ -1418,14 +1418,22 @@ void beam_render_muzzle_glow(beam *b)
 		vertex h1[4];
 		vertex *verts[4] = { &h1[0], &h1[1], &h1[2], &h1[3] };
 		vec3d fvec, top1, top2, bottom1, bottom2, sub1, sub2, start, end;
+		int idx;
+		float g_length = bwi->glow_length * pct * rand_val;
 
 		vm_vec_sub(&fvec, &b->last_shot, &b->last_start);
 		vm_vec_normalize_quick(&fvec);
-
-		vm_vec_copy_scale(&sub1, &fvec, rad);
-		vm_vec_sub(&start, &b->last_start, &sub1);
-		vm_vec_copy_scale(&sub2, &fvec, bwi->glow_length);
-		vm_vec_add(&end, &start, &sub2);
+		
+		if (bwi->glow_length >= 2.0f*rad) {
```

Comment:
This condition could use a comment to explain why the different behaviors are necessary. I assume that it has to do with how directional glows are handled but without looking through the entire code to determine what values `glow_length` and `rad` can take it is not possible to understand this condition.